### PR TITLE
update slack webhook with tri-state verification

### PR DESCRIPTION
**Pull request by zubairk14**
### Description:
Explain the purpose of the PR.

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/slackwebhook/slackwebhook_test.go [hunk 1]
```diff
@@ -94,9 +132,12 @@ func TestSlackWebhook_FromChunk(t *testing.T) {
 				if len(got[i].Raw) == 0 {
 					t.Fatalf("no raw secret present: \n %+v", got[i])
 				}
-				got[i].Raw = nil
+				if (got[i].VerificationError != nil) != tt.wantVerificationErr {
+					t.Fatalf(" wantVerificationError = %v, verification error = %v", tt.wantVerificationErr, got[i].VerificationError)
```
Comment (rosecodym) [inline pkg/detectors/slackwebhook/slackwebhook_test.go:136]: wherever you copied this from has a typo in the message (a leading space)

File: pkg/detectors/slackwebhook/slackwebhook.go [hunk 2]
```diff
@@ -12,7 +13,9 @@ import (
 	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
 )
 
-type Scanner struct{}
+type Scanner struct {
+	client *http.Client
```
Comment (rosecodym) [inline pkg/detectors/slackwebhook/slackwebhook.go:17]: `FromData` isn't actually using this client. this should be causing your new tests to fail - are you seeing failures locally? (for some reason github isn't pulling up the ci logs.)

File: pkg/detectors/slackwebhook/slackwebhook.go [hunk 3]
```diff
@@ -62,8 +65,18 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 					continue
 				}
 				body := string(bodyBytes)
-				if (res.StatusCode >= 200 && res.StatusCode < 300) || (res.StatusCode == 400 && (strings.Contains(body, "no_text") || strings.Contains(body, "missing_text"))) {
-					s1.Verified = true
+
+				if err == nil {
+					defer res.Body.Close()
+					if res.StatusCode >= 200 && res.StatusCode < 300 || (res.StatusCode == 400 && (strings.Contains(body, "no_text") || strings.Contains(body, "missing_text"))) {
```
Comment (rosecodym) [inline pkg/detectors/slackwebhook/slackwebhook.go:71]: out of curiosity, do you know what these magic strings are for? do they factor into the question of whether the candidate secret is live?

**Review by rosecodym — CHANGES_REQUESTED** [review]

File: pkg/detectors/slackwebhook/slackwebhook.go (hunk 2 above)
Comment (zubairk14) [reply pkg/detectors/slackwebhook/slackwebhook.go:17]: the tests are passing locally
image

File: pkg/detectors/slackwebhook/slackwebhook.go (hunk 3 above)
Comment (zubairk14) [reply pkg/detectors/slackwebhook/slackwebhook.go:71]: not sure myself - saw that they were part of the pre-existing verification logic and that it was written by a contractor so didn't have someone i could DM to ask

File: pkg/detectors/slackwebhook/slackwebhook.go (hunk 2 above)
Comment (zubairk14) [reply pkg/detectors/slackwebhook/slackwebhook.go:17]: image

but we are using *an* http.Client right? and we need to inject the timeout

```
s:    Scanner{client: common.SaneHttpClientTimeOut(1 * time.Microsecond)},
```

and the unexpected surface

```
s:    Scanner{client: common.ConstantResponseHttpClient(404, "")},
```
image

**Review by rosecodym — APPROVED** [review]
nice work!

**Pull request merged**